### PR TITLE
Always return true when validating DIDs with methods.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: lib/did-io.js [hunk 1]
```diff
@@ -187,6 +187,11 @@ api.generateHash = function(email, passphrase) {
...
  */
 api.validateDidFormat = function(did) {
+  // TODO: implement validation of specific methods
+  // for now, any did:foo:<string> is valid
+  if((did.match(/\:/g) || []).length === 2) {
+    return true;
+  }
   var didRegex = new RegExp('^did\:[0-9a-f]{8}-[0-9a-f]{4}-' +
     '4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$', 'i');
   return didRegex.test(did);
```
Comment (davidlehn) [inline lib/did-io.js]: I'd suggest pulling out the static regex setup and using both an old uuid style and new spec method style regex.  Just check the start part of the did for the new style.  Some lightly tested code:
```js
// old did:uuidv4 style
const didUuidRegex = new RegExp('^did\:[0-9a-f]{8}-[0-9a-f]{4}-' +
￼      '4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$', 'i');
// start of generic did scheme before path and fragment
// FIXME: accepts bad trailing ":" before path: "did:method:specific:idstring:/path
const didGenericRegex = /^did:([a-z0-9]+):([A-Za-z0-9.-]+(?::[A-Za-z0-9.-]+)*)/
api.validateDidFormat = function(did) {
  return didUuidRegex.test(did) || didGenericRegex.test(did);
￼}
￼```

Comment (dlongley) [reply lib/did-io.js]: +1 for using the generic DID regex from the new DID spec.

**Comment by mattcollier** [comment]
@dlongley @davidlehn ready for another pass.  I'm not sure if the FIXME included in the code lehn provided should be fixed now or later.

**Comment by davidlehn** [comment]
I think the regexes should be outside the `api.validateDidFormat` function since they don't need to be recreated on each call.  I'd suggest just leaving the FIXME in there unless someone wants to fix the regex for the one edge case false positive.

**Comment by mattcollier** [comment]
@davidlehn ready for another pass.

**Review by dlongley — APPROVED** [review]

**Pull request merged**